### PR TITLE
Phase2-hgx124 Split the SD for HGCal for silicon and scintillators

### PR DESCRIPTION
**Pull request by bsunanda**
Allow fiducial volume cuts in silicon and scintillator detectors of HGCal

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/Calo/src/HGCalSD.cc [hunk 1]
```diff
@@ -215,26 +212,24 @@ void HGCalSD::update(const BeginOfJob * job) {
...
 			       << " top Level " << levelT1_ << ":" << levelT2_ 
-			       << " isScint " << isScint_ << " wafer " 
-			       << waferSize << ":" << mouseBite;
+			       << " wafer " << waferSize << ":" << mouseBite;
 #endif
 
-    numberingScheme_ = new HGCalNumberingScheme(*hgcons,mydet_,nameX_);
-    if (rejectMB_ && (!isScint_)) 
-      mouseBite_ = new HGCMouseBite(*hgcons,angles_,mouseBiteCut_,waferRot_);
+    numberingScheme_ = new HGCalNumberingScheme(*hgcons_,mydet_,nameX_);
```
Comment (kpedro88) [inline SimG4CMS/Calo/src/HGCalSD.cc]: smart pointers here as well

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 2]
```diff
@@ -242,10 +243,88 @@ double HGCalDDDConstants::cellSizeHex(int type) const {
   return cell;
 }
 
+double HGCalDDDConstants::distFromEdgeHex(double x, double y, double z) const {
```
Comment (ianna) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda - please, add a comment what is calculated here. Is the point(x,y) inside of the polygon?

**Review by ianna — CHANGES_REQUESTED** [review]

**Comment by bsunanda** [comment]
Hi Ianna

This is for a point inside the volume. I shall add a comment for the next version which is under preparation right now. Could this PR be merged as it is now. That will help in the progress of the next PR.  Best regards

Sunanda
________________________________

**Comment by bsunanda** [comment]
@ianna @civanch Can you approve this PR so that I can make the subsequent PR's

**Comment by civanch** [comment]
+1

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc (hunk 2 above)
Comment (ianna) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda it takes longer for me to check the code without comment on what it suppose to do.

**Comment by fabiocos** [comment]
+1

it affects only Phase2, g4SimHits is changed only for SD of HGCal.

**Pull request merged**